### PR TITLE
Build zbar-v0.23.92

### PR DESCRIPTION
**Pull request by RexWzh**
Build script for zbar-v0.23.92.

I have tested the platforms listed in `supported_platforms()` in my machine, all platforms passed except for `x86_64-linux-gnu`. (It works fine in auto-test.)
```bash
julia build_tarballs.jl --debug --verbose xxx-xxx-xxx
```
The error message is

```bash
make[2]: Entering directory '/workspace/srcdir/zbar'
/bin/sh ./libtool  --tag=CC   --mode=link cc -std=gnu99 -Wall -Wno-parentheses -DLOCALEDIR=\"/workspace/destdir/share/locale\" -g -O2   -o zbarimg/zbarimg zbarimg/zbarimg-zbarimg.o  zbar/libzbar.la -L/workspace/destdir/lib -lMagickWand-6.Q16 -lMagickCore-6.Q16   -lpthread -lrt 
libtool: link: cc -std=gnu99 -Wall -Wno-parentheses "-DLOCALEDIR=\"/workspace/destdir/share/locale\"" -g -O2 -o zbarimg/.libs/zbarimg zbarimg/zbarimg-zbarimg.o  zbar/.libs/libzbar.so -L/workspace/destdir/lib -lMagickWand-6.Q16 -lMagickCore-6.Q16 -lpthread -lrt -Wl,-rpath -Wl,/workspace/destdir/lib
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_memset'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_memcpy'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_retval_tls'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_va_arg_overflow_size_tls'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_memmove'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_va_arg_tls'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_warning_with_origin_noreturn'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_init'
/workspace/destdir/lib/libz.so.1: undefined reference to `__msan_param_tls'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1359: zbarimg/zbarimg] Error 1
make[2]: Leaving directory '/workspace/srcdir/zbar'
make[1]: *** [Makefile:1897: all-recursive] Error 1
make[1]: Leaving directory '/workspace/srcdir/zbar'
make: *** [Makefile:977: all] Error 2
```

Version info:
```julia
julia> versioninfo()
Julia Version 1.6.3
Commit ae8452a9e0 (2021-09-23 17:34 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, icelake-client)
```

Update: Use the same platforms as `ImageMagick`.

### CODE REVIEW

**Comment by RexWzh** [comment]
Well, it looks fine in auto-test. :)
Might have to do with my local machine?

File: Z/zbar/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            --prefix=${prefix} \
+            --build=${MACHTYPE} \
+            --host=${target}
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(;experimental=true))
```
Comment (giordano) [inline Z/zbar/build_tarballs.jl:39]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: Z/zbar/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="ImageMagick_jll", uuid="c73af94c-d91f-53ed-93a7-00f77d67a9d7"))
+    Dependency(PackageSpec(name="Libiconv_jll", uuid="94ce4f54-9a6c-5748-9c1c-f9c7231a4531"); platforms=filter(!Sys.isapple, platforms))
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline Z/zbar/build_tarballs.jl:53]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")

```

Comment (RexWzh) [reply Z/zbar/build_tarballs.jl]: Thanks for taking your time to review the code! I think the build script might be fine now.

File: Z/zbar/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("zbarimg", :zbarimg),
```
Comment (giordano) [inline Z/zbar/build_tarballs.jl:43]: You only need the executable product?  Not the shared library?

Comment (RexWzh) [reply Z/zbar/build_tarballs.jl:44]: Thank you for the reminder, this is my first time doing this process, I am not very familiar with it yet. I would build and test JLL packages locally first to make sure they can be called correctly.

**Comment by giordano** [comment]
Was there any problem?

**Comment by RexWzh** [comment]
Sorry, my intention was to change it to draft, but mis-clicked it on my phone.

The artifact downloaded from auto-test works fine through JLL package, and can be used directly after downgrading the local ImageMagick from 7.0 to 6.9. (It seems that [ImageMagick_jll](https://github.com/JuliaBinaryWrappers/ImageMagick_jll.jl) has not been updated to version 7.)

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**